database/memorydb.go

```diff
@@ -0,0 +1,81 @@
+package database
+
+import (
+	"fmt"
+	"sync"
+)
+
+type todos map[int]string
+
+type Database struct {
+	mu     sync.Mutex
+	data   todos
+	lastId int
+}
+
+func CreateDatabase() *Database {
+	return &Database{data: make(todos), lastId: -1}
+}
+
+func (db *Database) FetchTodo(id int) (string, error) {
+	var err error
+	db.mu.Lock()
+	defer db.mu.Unlock()
+
+	val, exists := db.data[id]
+
+	if !exists {
+		err = fmt.Errorf("ID: %v doesn't exist", id)
```

Comment:
Avoid mutables, make a return here

Reply:
Never start errors with capital letters, read the Go formatting guide and the tour of Go.

Reply:
Added return 6762dd75ebabb0f65357d3b82550d003cab556d4 and fixed error formatting b01288b49a50ddb360137ebdbfc85520895c183f